File: docs/js/api/ratings.js
"use strict";
import { BASE_URL, requestOptions } from "/js/api/common.js";

const ratingsAPI = {
    getSubmittedScore: function(photoId,userId){
        return new Promise(function(resolve,reject){
            axios.get(BASE_URL + "/score/photos/"+ photoId +"/"+userId)
                .then(response => resolve(response.data[0]))
                .catch(error => reject(error.response.data.message));
        });
    },
    create: function (formData) {
        return new Promise (function(resolve,reject){
            axios.post(BASE_URL + "/ratings", formData)
                .then(response => resolve (response.data))
                .catch(error => reject (error.response.data.message));
        });
    },
};


export { ratingsAPI };

File: endpoints/ratings.py
from silence.decorators import endpoint

@endpoint(
    route="/score/photos/$photoId/$userId",
    method="GET",
    sql="SELECT * FROM Ratings WHERE photoId=$photoId AND userId=$userId"
)
def get_submitted_score():
    pass

###############################################################################

@endpoint(
    route="/ratings",
    method="POST",
    sql="INSERT INTO Ratings(score,userId,photoId) VALUES ($score, $userId, $photoId) ON DUPLICATE KEY UPDATE score=$score",
)
def create(score,userId,photoId):
    pass